libraries/engage/push-opt-in/components/PushOptInModal/index.jsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Grid, I18n, Button, Modal,
+} from '@shopgate/engage/components';
+import pushImage from './push-opt-in.svg';
+import styles from './style';
+import connect from './connector';
+
+/**
+ * The Push opt-in modal component.
+ * @param {Object} props The component props.
+ * @returns {JSX.Element}
+ */
+const PushOptInModal = ({
+  showPushOptInModal, enablePushOptInModal, denyPushOptInModal, hidePushOptInModal,
+}) => {
+  /**
+   * opens native modal to give the permission for sending push notifications
+   * and closes the custom modal
+   */
+  const handleClickEnable = () => {
+    enablePushOptInModal();
+    hidePushOptInModal();
+  };
+
+  /**
+   * Denies the permission for sending push notifications and closes the modal
+   */
+  const handleClickDeny = () => {
+    denyPushOptInModal();
+    hidePushOptInModal();
+  };
+
+  if (!showPushOptInModal) {
+    return null;
+  }
+
+  return (
+    <Modal isOpened={showPushOptInModal}>
+      <Grid className={styles.container}>
+        <Grid.Item className={styles.item}>
+          <img src={pushImage} className={styles.image} alt="push opt-in graphic" />
```

Comment:
Is there a reason why "modalImage", "modalTitle" etc have a value prop?
I Assume this was copied from "ordersPlaced" and "appStarts" but i think we dont need it here.
In addition we need the defaults in the extension-config.

Reply:
yes, I wasn't sure how the config should look like exactly, but it worked like that. So instead of value I would just put "modalImage": null ? It's already in the default section of the pushOptin, so should be correct, yes?

Reply:
fixed as discussed